demo/d2l-input-checkbox.html

```diff
@@ -97,6 +97,13 @@ <h3>Checkbox with label and secondary content</h3>
 				</template>
 			</demo-snippet>
 
+			<h3>Indeterminate checkbox</h3>
+			<demo-snippet>
+				<template>
+					<d2l-input-checkbox checked indeterminate>Indeterminate checkbox</d2l-input-checkbox>
```

Comment:
If I want tri-state behavior, but initially it's in an unchecked state, what mark-up would I use.  Does `indeterminate` control the behavior, or reflect the current state?

Reply:
You'd just exclude the `checked` attribute. Reading through the specs, `indeterminate` is purely stylistic -- it has no functional or behavioural semantics. ARIA seems to add a bit there, but as far as the browsers know, it's just a weird style thing.

Reply:
Does this assume the user cannot click to return to the mixed state?  The keyboard support comments [here](https://www.w3.org/TR/wai-aria-practices/examples/checkbox/checkbox-2/checkbox-2.html#kbd_label) indicate that this should be possible, which would imply that the click/change handler would need to be different.

Reply:
So I believe that's just an example of a best practice using tri-state checkbox to create that particular experience -- one with sub-items where the number of child items that are checked corresponds with the state of the tri-state checkbox. That's just one implementation though... so I don't think we'd want to bake that particular behaviour in.

I would imagine that the consumers of this would programatically toggle the `indeterminate` attribute on the parent checkbox based on how many child checkboxes are checked. Checking and unchecking the parent might subsequently check and un-check all the children. I can't think of why in this particular case you'd want to cycle through the indeterminate state on the parent by clicking it though... seems weird.

Reply:
Yeah, in this case, the parent would just set the "indeterminate", because it's based on whether its children are clicked. If you click on the indeterminate state, then it will go to the **opposite** of either "checked" or "unchecked" (whichever was set while it was indeterminate). You can't get back to the indeterminate state by clicking on the same checkbox. You would have to click on its children and the parent would set it again.

Reply:
The only use-case I can think of for supporting this is the ability to "undo".  Currently, a user would either have recheck/uncheck the appropriate items, or reload the page (assuming no-auto-save!) if they accidentally click.

Reply:
^ I agree that would be really handy, but so far it is not in our workflow. If any team needed that functionality, they could add it :D